vac/raw/consensus-hashgraphlike.md

```diff
@@ -0,0 +1,219 @@
+---
+## Abstract
+
+This document specifies a scalable, decentralized, and Byzantine Fault Tolerant (BFT)
+consensus mechanism inspired by Hashgraph, designed for binary decision-making in P2P networks.
+
+## Motivation
+
+Consensus is one of the essential components of decentralization.
+In particular, in the decentralized group messaging application is used for
+
+binary decision-making to govern the group.
+Therefore, each user contributes to the decision-making process.
+Besides achieving decentralization, the consensus mechanism MUST be strong:
+under the assumption of at least 2/3 honest users in the network,
+each user MUST conclude the same decision and scalability:
+the rounds MUST be growth logarithmic so as not to lose the scalability of the messaging applications.
+
+## Format Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”,
+“SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document
+are to be interpreted as described in [2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## Flow
+
+Any user in the group initializes the consensus by creating a proposal.
+Next, the user broadcasts the proposal to the whole network.
+Upon each user receives the proposal, validates the proposal,
+adds its vote as yes or no and with its signature and timestamp.
+Then this user sends the vote to the random node in the network or subscribing channel.
+Therefore, each user first validates the signature and then adds its new vote.
+Each sending message counts as a round.
+After log(n) rounds all users in the network have the others vote
+if at least 2/3 number of users are honest where honesty follows the protocol.
+
+In general, the voting-based consensus consists of the following phases:
+
+1. Initialization of voting
+2. Exchanging votes across the rounds
+3. Counting the votes
+
+### Assumptions
+
+- The users in the P2P network can discover the nodes or they are subscribing same channel in a gossipsub.
+- We MAY have non-reliable (silent) nodes.
+- Proposal owners MUST know the number of voters.
+
+## 1. Initialization of voting
+
+A user initializes the voting with the proposal payload which is
+implemented using [protocol buffers v3](https://protobuf.dev/) as follows:
+
+```bash
+syntax = "proto3";
+
+package vac.voting;
+
+message Proposal {
+  string name = 10;                 // Proposal name
+  int32 proposal_id = 11;           // Unique identifier of the proposal
+  bytes proposal_owner = 12;        // Public key of the creator 
+  repeated Votes = 13;              // Vote list in the proposal
+  int32 expected_voters_count = 14; // Maximum number of distinct voters
+  int32 round = 15;                 // Number of Votes 
+  int64 timestamp = 16;             // Creation time of proposal
+  int64 expiration_time = 17;       // The time interval that the proposal is active.  
+  bool silent_node_yes = 18;       // Shows how managing the silent peers vote

```

Comment:
You refer to `liveness_criteria_yes`in silent node section, but here it's still `silent_node_yes`